extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/VertxRequestHandler.java

```diff
@@ -95,10 +144,83 @@ public void handle(RoutingContext routingContext) {
             log.error("Illegal media type:" + mediaType);
             return;
         }
+
     }
 
-    private static final ResponseProcessing NOOP = () -> {
-    };
+    private void processCloudEvent(RoutingContext routingContext) {
+        final HttpServerRequest request = routingContext.request();
+        final String mediaType = request.getHeader("Content-Type");
+        final boolean binaryCE = request.headers().contains("Ce-Id");
+        final Consumer<Throwable> fail = t -> routingContext.fail(t);
+
+        messageReader(routingContext)
+                .emitOn(executor)
+                .subscribe()
+                .with(messageReader -> {
+                    try {
+                        CloudEvent inputCE = messageReader.toEvent();
+
+                        final String ceType = inputCE.getType();
+                        final FunctionInvoker invoker;
+                        if (defaultInvoker != null) {
+                            invoker = defaultInvoker;
+                        } else {
+                            invoker = typeTriggers.get(ceType);
+                            if (invoker == null) {
+                                routingContext.fail(404);
+                                log.error("Could CloudEvent type: " + ceType + " to a function");
+                                return;
+                            }
+                        }
+
+                        Object inputData = null;
+                        if (invoker.hasInput()) {
+                            PojoCloudEventDataMapper<?> ceDataMapper = PojoCloudEventDataMapper.from(mapper,
```

Comment:
Use `CloudEventUtils.mapTo` (just merged on sdk-java master, snapshots will be deployed soon)